### ISSUE TITLE
Listing function bug

### ISSUE THREAD

**Issue by janvanrijn** [openml/openml-python#463]
#### Description
On the pip version of OpenML I keep on getting the following error:
```
Traceback (most recent call last):
  File "/home/janvanrijn/projects/openml-aslib/examples/generate_from_study.py", line 22, in <module>
    require_complete=args_.require_complete)
  File "/home/janvanrijn/projects/openml-aslib/openmlaslib/utils/scenario.py", line 10, in generate_scenario
    function=measure, setup=setups, task=tasks)
  File "/home/janvanrijn/anaconda3/envs/openml-aslib/lib/python3.6/site-packages/openml/utils.py", line 147, in list_all
    **active_filters
TypeError: list_evaluations() got an unexpected keyword argument 'limit'
```

#### Steps/Code to Reproduce

```python
import openml
evaluations = openml.utils.list_all(openml.evaluations.list_evaluations,
                                        function='predictive_accuracy', setup=[1], task=[1])
```

#### Notes
I checked into the code, there are several things that I noted:
* There are practically no unit tests for this functionality
* The unit test that is there works on basis of a hidden function (i.e., _list_datasets)
* There are several hidden functions for the list setup function? _list_setups and __list_setups?
* When adding unit tests on the correct function, they practically all fail with with either of the following errors:
** TypeError: _list_flows() got multiple values for keyword argument 'limit'
** TypeError: list_evaluations() got an unexpected keyword argument 'limit'

**Comment by janvanrijn** [comment]
Apparently, I should just call the `list_datasets` etc function, which in turn utilizes the list all function. Shouldn't we make the list all function hidden? Otherwise people might use it, whereas apparently this is not directly intended to be used. 

Still I get errors:
```
    def test_list_all_for_tasks(self):
        tasks = openml.tasks.list_tasks(size=100000)

        self.assertGreaterEqual(len(tasks), 500)
        self.assertLessEqual(len(tasks), 100000)
```

(returns empty dict)

**Comment by janvanrijn** [comment]
Apparently this can be fixed by changing the fn call into the following.
```
new_batch = listing_call(
                *args,
                limit=batch_size,
```

Additionally, things go wrong with the batch size parameter. When I request a size that is not a multiplier of the batch_size, I don't get the exact amount. 

I would also catch the error (in case of no results) in a different way. I will make a PR.

**Comment by janvanrijn** [comment]
Luckily, I was wrong. The functions did not contain the aforementioned bug, as I invoked the wrong function. (Apparently the behavior has changed in the meantime, which is fine by me :) )

I added unit tests and unfortunately found another (less serious) bug. The returned number of the listing does not always return the correct value, it sometimes stops too early. I have fixed this. 

Furthermore, I think that it is confusing to give the impression that this function is publicly callable, as it requires to invoke private functions. As such, I made this function private as well (underscore naming convention). 

Is there something we can do about the triple occurrence of functions? i.e., we now have list_datasets, _list_datasets, __list_datasets and _list_all..
